package.json

```diff
@@ -1,12 +1,20 @@
 {
   "private": true,
   "license": "MIT",
-  "workspaces": [
-    "contracts",
-    "operator_ui",
-    "tools",
-    "tools/external-adapter"
-  ],
+  "workspaces": {
+    "packages": [
+      "contracts",
+      "operator_ui",
+      "tools",
+      "tools/external-adapter"
+    ],
+    "nohoist": [
+      "**/typechain",
+      "**/typechain/**",
+      "**/@typechain/ethers-v5",
+      "**/@typechain/ethers-v5/**"
+    ]
```

Comment:
typechain and @typechain/ethers-v5 have some kind of peer dependency with a hard-coded path. They won't work unless they're in the same node_modules folder. Yarn was hoisting one, but no the other. This config forces them to be in the same location.